simulators/kcc/devp2p-eth66/main.go

```diff
@@ -0,0 +1,354 @@
+			//isSyncToNewNode = true
+			isSyncBlock = isSyncBlock | 2
+		}
+		t.Logf("[client1] get block, %d, validator: %s", block.Number().Uint64(), block.Header().Coinbase.String())
+
+		if isSyncBlock == 3 {
+			break
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	if isSyncBlock != 3 {
+		t.Fatalf("sync node error, %d", isSyncBlock)
+	}
+
+	t.Log("sync block success, between old and new node")
+
+	// ################# transaction sync test #####################
+	validatorAddress := []string{"0x658bdf435d810c91414ec09147daa6db62406379", "0x7B42a4573ECb800bD404042F518a59CfBE3b582B", "0x7f0dd90F50E4E18D58aD7780a5a642acd5f44068"}
+
+	newHead := make(chan *types.Header, 16)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	clients := []*ethclient.Client{oldEthClient, newEthClient1, newEthClient2}
+	url := fmt.Sprintf("ws://%v:8546", oldClient.IP)
+	wsClient, err := rpc.Dial(url)
+	if err != nil {
+		t.Fatalf("new ws client error, url:%s, err:%v", url, err)
+	}
+
+	opt, err := ethclient.NewClient(wsClient).SubscribeNewHead(ctx, newHead)
+	if err != nil {
+		t.Fatalf("sub new head err, err:%v", url, err)
+	}
+
+	isSync := 0
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			t.Fatalf("watch new head time out")
+
+		case header := <-newHead:
+			b, _ := header.MarshalJSON()
+			t.Log("new head", string(b))
+
+			for i, address := range validatorAddress {
+				if strings.ToLower(address) != strings.ToLower(header.Coinbase.String()) {
+					continue
+				}
+
+				switch i {
+				case 0: // old node send transaction sync to new node
+					sendTransaction(t, validations[0].privateKey, oldEthClient, clients)
+					isSync = isSync | 1
+				//case 1: // new node send transaction sync to new node and old node
+				//	sendTransaction(t, validations[1].privateKey, newEthClient1, clients)
+				//	isSync = isSync | 2
+				case 2: // new node send transaction sync to new node and old node
+					sendTransaction(t, validations[2].privateKey, newEthClient2, clients)
+					isSync = isSync | 2
+				}
+			}
+
+			if isSync == 3 {
+				break LOOP
+			}
+		}
+	}
+	opt.Unsubscribe()
+
+	t.Log("sync transaction success, between old and new node")
+}
+
+func sendTransaction(t *hivesim.T, privateKey string, client *ethclient.Client, clients []*ethclient.Client) {
+	pk, err := crypto.HexToECDSA(privateKey)
+	if err != nil {
+		t.Fatal("HexToECDSA err:", err)
+	}
+
+	signer := types.LatestSignerForChainID(big.NewInt(321))
+	address := common.HexToAddress(validations[0].miner)
+	transaction := &types.LegacyTx{
+		GasPrice: big.NewInt(2 * 1e9),
+		Gas:      8 * 1e5,
+		To:       &address,
+		Value:    big.NewInt(0),
+	}
+
+	tx, err := types.SignNewTx(pk, signer, transaction)
+	if err != nil {
+		t.Fatal("sign new tx error", err)
+	}
+
+	err = client.SendTransaction(context.TODO(), tx)
+	if err != nil {
+		t.Fatal("send transaction error", err)
+	}
+
+	//after 50ms to sync
+	time.Sleep(time.Millisecond * 50)
```

Comment:
Aha,  I think there is a better way to do this. You can stop the miners (with RPC call `miner_stop`) before broadcasting the txs.